src/DssSpell.t.sol

```diff
@@ -646,40 +628,97 @@ contract DssSpellTest is DSTest, DSMath {
+       scheduleWaitAndCastFailEarly();
+   }
+
+   function testFailTooLate() public {
+       vote();
+       scheduleWaitAndCastFailLate();
+   }
+
+   function testOnTime() public {
+       vote();
+       scheduleWaitAndCast();
+   }
+
+    // function testSpellIsCast() public {
+    //     string memory description = new DssSpell().description();
+    //     assertTrue(bytes(description).length > 0);
+    //     // DS-Test can't handle strings directly, so cast to a bytes32.
+    //     assertEq(stringToBytes32(spell.description()),
+    //             stringToBytes32(description));
+
+    //     if(address(spell) != address(MAINNET_SPELL)) {
+    //         assertEq(spell.expiration(), (now + 4 days + 2 hours));
+    //     } else {
+    //         assertEq(spell.expiration(), (SPELL_CREATED + 4 days + 2 hours));
+    //     }
+
+    //     vote();
+    //     scheduleWaitAndCast();
+    //     assertTrue(spell.done());
+
+    //     checkSystemValues(afterSpell);
+
+    //     checkCollateralValues(afterSpell);
+    // }
+
+    function test_nextCastTime() public {
+        hevm.warp(1606161600); // Nov 23, 20 UTC (could be cast Nov 26)
 
         vote();
-        scheduleWaitAndCast();
-        assertTrue(spell.done());
+        spell.schedule();
+
+        uint256 monday_1400_UTC  = 1606744800; // Nov 30
+
+        /*** Day tests */
+        hevm.warp(monday_1400_UTC); // Monday, 14:00 UTC
+        assertEq(spell.nextCastTime(), monday_1400_UTC);
+
+        hevm.warp(monday_1400_UTC - 1 days); // Sunday, 14:00 UTC
+        assertEq(spell.nextCastTime(), monday_1400_UTC);
+
+        hevm.warp(monday_1400_UTC - 2 days); // Saturday, 14:00 UTC
+        assertEq(spell.nextCastTime(), monday_1400_UTC);
+
+        hevm.warp(monday_1400_UTC - 3 days); // Friday, 14:00 UTC
+        assertEq(spell.nextCastTime(), monday_1400_UTC - 3 days); // Able to cast
+
+        /*** Time tests ***/
+        uint256 castTime;
+
+        for(uint i = 0; i < 5; i++) {
+            castTime = monday_1400_UTC + i * 1 days; // Next day at 1400 UTC
+            hevm.warp(castTime - 1 seconds); // 13:59 UTC
+            assertEq(spell.nextCastTime(), castTime);
 
-        checkSystemValues(afterSpell);
+            hevm.warp(castTime + 7 hours + 1 seconds); // 21:01 UTC
+            if (i < 4) {
+                assertEq(spell.nextCastTime(), monday_1400_UTC + (i + 1) * 1 days); // Next day at 14:00 UTC
+            } else {
+                assertEq(spell.nextCastTime(), monday_1400_UTC + 7 days); // Next monday at 14:00 UTC (friday case)
+            }
+        }
+    }
+
+    function testFail_notScheduled() public {
+        spell.nextCastTime();
+    }
+
+    function test_use_eta() public {
+        hevm.warp(1606161600); // Nov 23, 20 UTC (could be cast Nov 26)
+
+        vote();
+        spell.schedule();
 
-        checkCollateralValues(afterSpell);
+        uint castTime = spell.nextCastTime();
+        assertEq(castTime, spell.eta()); // Nov 26, 20 UTC
```

Comment:
If we are planning to leave this test as definitive the comment shouldn't say `Nov 26, 20 UTC` but `Nov 23, 20 UTC + pause.delay()` as it won't be always `72 hours`.

Reply:
Ah right, I'll just delete the comment.